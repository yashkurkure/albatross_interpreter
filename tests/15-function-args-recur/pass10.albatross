var result int := 0;
var n int := 0;

fun factorial int (i int) {
  if (i == 0) { return 1; }
  else {
    result := i * factorial(i-1);
    return result;
  }
}

repeat(15) {
    factorial(n);
    printint(result);
    printstring(" ");
    n := n+1;
}
return 0;
