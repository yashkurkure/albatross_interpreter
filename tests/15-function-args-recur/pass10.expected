0 1 2 6 24 120 720 5040 40320 362880 3628800 39916800 479001600 1932053504 1278945280 
